testsuite/tests/comprehensions/list_comprehensions_modules.ml

```diff
@@ -0,0 +1,43 @@
+(* TEST
+  flags = "-extension comprehensions_experimental"
+   * expect
+*)
+
+module type S = sig
+  type t
+  val x : t
+end;;
+
+let t = (module struct
+  type t = int
+  let x = 3
+end : S);;
+[%%expect {|
+module type S = sig type t val x : t end
+val t : (module S) = <module>
+|}];;
+
+let f () =
+  [ M.x for (module M : S) in [ t ] ];;
+[%%expect {|
+Line 2, characters 20-21:
+2 |   [ M.x for (module M : S) in [ t ] ];;
+                        ^
+Error: Modules are not allowed in this pattern.
+|}];;
+
+let f () =
+  [ M.x
+    for (module M :S) in
+      [ (module struct
+          type t = int
+          let x = 3
+         end : S)
+      ]
+  ];;
+[%%expect {|
+Line 3, characters 16-17:
+3 |     for (module M :S) in
+                    ^
+Error: Modules are not allowed in this pattern.
+|}];;
```

Comment:
Can you, instead of creating this file, add these tests to all of `{list,array,iarray}_comprehensions_pure.ml`?  They could go nicely at the end of the "`(**** Errors ****)`" section.

(If you'd really prefer it as a separate file, I'd rename it to `comprehensions_module_patterns.ml` and duplicate these tests for both array and immutable array comprehensions, but I think adding them next to the existing tests is cleaner, personally.)

Reply:
This is done.